pkg/proxy/iptables/proxier.go

```diff
@@ -132,7 +132,7 @@ type endpointsInfo struct {
 func newEndpointInfo(baseInfo *proxy.BaseEndpointInfo, svcPortName *proxy.ServicePortName) proxy.Endpoint {
 	return &endpointsInfo{
 		BaseEndpointInfo: baseInfo,
-		ChainName:        servicePortEndpointChainName(svcPortName.String(), strings.ToLower(string(svcPortName.Protocol)), baseInfo.Endpoint),
+		ChainName:        servicePortEndpointChainName(svcPortName.String(), strings.ToLower(string(svcPortName.Protocol)), baseInfo.Endpoint.String()),
```

Comment:
I have doubts this improves performance, since we need to concatenate IP port in each of these calls

Reply:
Yes, currently the `endpointString` struct only stores the IP and Port, so the `proxier` doesn't need to perform additional parsing to extract them. However, for the `String` method, concatenation is needed.

I can add an endpoint field to the endpointString struct:
```go
type endpointString struct {
	IP       string
	Port     int
	endpoint string
}

func (e *endpointString) String() string {
	if e.endpoint == "" {
		e.endpoint = net.JoinHostPort(e.IP, strconv.Itoa(e.Port))
	}
	return e.endpoint
}
```

What do you think?

Reply:
I just saw you claimed the performance improvement, and I was wondering how we can know if we regress or improve, without that information is hard to evaluate if this change is necessary

Reply:
https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/iptables/proxier.go#L1371

https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/ipvs/proxier.go#L1051 

At least here, there is no need to repeatedly extract the Endpoint string when calling the `IP()` and `Port()` methods.

Do you have any suggestions for performance improvement testing?